### PR TITLE
Feature/i18n profiles

### LINKED ISSUE #797: Deliver the right extension & profiles app version depending on the language of the user's browser (English Version)
Implement a library to deliver the right translation depending on the language version of the browser

- If English, use English translations
- If French, use French translations
- If any other language, use English translations

### LINKED ISSUE #810: Prepare Profiles app components to be translated

### LINKED ISSUE #812: Prepare i18n file for translation (French Version) for Profiles

### LINKED ISSUE #825: Make the translation of the "profiles" work on the storybook
The issue: we have to use 2 instances of i18next (extension and profile) in the same time on storybook. But it doesn't work. Only one at a time.
I suggest to keep only one I18next config file inside the locales root folder and use the same file for both (extension and profiles) app. 
We will use name spaces to handle the right translation inside the 2 apps. 

are you ok @lutangar @JalilArfaoui ?

### PR DESCRIPTION
**Pull request by gregoirelacoste**
closes #797 closes #812 closes #810 closes #825

### CODE REVIEW

**Comment by gregoirelacoste** [comment]
@christpet @lutangar the profiles app is ready to be translate. I finished adding all the function inside the components , could you review ? thanks !

File: src/app/profiles/i18n/index.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import i18n from 'i18next';
+import LanguageDetector from 'i18next-browser-languagedetector';
+import { initReactI18next } from 'react-i18next';
+import resources from '../../../locales/resources';
```
Comment (lutangar) [inline src/app/profiles/i18n/index.ts:4]:
```
import resources from 'locales/resources';
```
Should works.

File: src/app/profiles/i18n/index.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import format from './format';
+
+const options = {
+  resources,
```
Comment (lutangar) [inline src/app/profiles/i18n/index.ts:8]: We might find a way to share most options.

**Review by lutangar — CHANGES_REQUESTED** [review]

File: src/app/profiles/App/Pages/Profiles/NotConnectedPopin.tsx [hunk 3]
```diff
@@ -44,21 +45,25 @@ const NotConnectedPopin = ({
...
   addToBrowser
 }: NotConnectedPopinProps) => {
+  const { t } = useTranslation();
   return (
     <PopinLarge size={'extralarge'} opened={opened} setOpened={setOpened}>
       <Text>
-        Pour voir les contributions {contributor && `de ${contributor.name}`},
-        veuillez d’abord ajouter Dismoi à votre navigateur.
+        {t('popin.not_connected.intro', {
+          contributorName: contributor && ' de ' + contributor.name
```
[the remark this answers is not in the document]
Comment (gregoirelacoste) [reply src/app/profiles/App/Pages/Profiles/NotConnectedPopin.tsx:53]: if there is contributor, i need to keep also the string "de" before contributor.name

[the remark this answers is not in the document]
Comment (lutangar) [reply src/app/profiles/App/Pages/Profiles/NotConnectedPopin.tsx:53]: I think it's typically the kind of thing that should be done in the translation itself... I'll try to verify this.

File: src/app/profiles/App/Pages/Error.tsx [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
-    <p>
-      Quelque chose s’est mal passé. Veuillez actualiser le plugin ou réessayer
-      plus tard.
-    </p>
-  </div>
-);
+export default () => {
+  const { t } = useTranslation();
+  return (
+    <div>
```
[the remark this answers is not in the document]
Comment (gregoirelacoste) [reply src/app/profiles/App/Pages/Error.tsx:7]: @lutangar sorry, i do not see the difference ?

[the remark this answers is not in the document]
Comment (lutangar) [reply src/app/profiles/App/Pages/Error.tsx:7]: You're absolutely right! My bad...

File: src/app/profiles/App/Pages/Profiles/NotConnectedPopin.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (gregoirelacoste) [reply src/app/profiles/App/Pages/Profiles/NotConnectedPopin.tsx:53]: new method with context 
with t function on the component : 
```
t('popin.not_connected.intro', {
   contributorName: contributor?.name,
    context: contributor?.name && 'name'
})
```
on the translation file`:
```
"not_connected": {
      "intro": "Pour voir les contributions, veuillez d’abord ajouter Dismoi à votre navigateur.",
      "intro_name": "Pour voir les contributions de {{contributorName}}, veuillez d’abord ajouter Dismoi à votre navigateur.",
      ...
    },
```
ok for you ?

[the remark this answers is not in the document]
Comment (lutangar) [reply src/app/profiles/App/Pages/Profiles/NotConnectedPopin.tsx:53]: If it works, ofc I am!

File: src/app/profiles/i18n/format.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { format as dateFormat } from 'date-fns';
+
+type FormatValue = string | Date;
+
+export default function(value: FormatValue, format?: string, _lng?: string) {
```
[the remark this answers is not in the document]
Comment (gregoirelacoste) [reply src/app/profiles/i18n/format.ts:5]: i will do that in https://github.com/dis-moi/extension/issues/825

File: src/app/profiles/i18n/index.ts (hunk 2 above)
Comment (gregoirelacoste) [reply src/app/profiles/i18n/index.ts:8]: i will do that in #825

**Comment by gregoirelacoste** [comment]
@lutangar @christpet i reorganized the files to use only one config file at the root of src/local. I use namespace "profiles" for profiles app and configured "extensions" as default namespace

File: src/app/profiles/App/Pages/Profiles/Profile/ProfileBanner.tsx [hunk 6]
```diff
@@ -55,7 +60,7 @@ const ProfileBanner = ({ contributor }: ProfileBannerProps) => {
             ? contributor.banner
             : defaultProfileBannerImage
         }
-        alt={getBannerAlt(contributor)}
+        alt={getBannerAlt(t, contributor)}
```
[the remark this answers is not in the document]
Comment (lutangar) [reply src/app/profiles/App/Pages/Profiles/Profile/ProfileBanner.tsx]: This isn't resolved as far as I can? Could use context here.
Or maybe you forgot to push?

File: src/app/profiles/App/Pages/Profiles/Profile/ProfileBanner.tsx [hunk 7]
```diff
@@ -50,7 +49,7 @@ const ProfileBanner = ({ contributor }: ProfileBannerProps) => {
             ? contributor.banner
             : defaultProfileBannerImage
         }
-        alt={t('view.profile.banner', {
+        alt={t('profiles:view.profile.banner', {
```
Comment (gregoirelacoste) [inline src/app/profiles/App/Pages/Profiles/Profile/ProfileBanner.tsx:52]: @lutangar  the last change is here

File: src/app/profiles/App/Pages/Profiles/Profile/ProfileBanner.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (gregoirelacoste) [reply src/app/profiles/App/Pages/Profiles/Profile/ProfileBanner.tsx]: this file seems to be outdated

**Review by lutangar — CHANGES_REQUESTED** [review]
Looking good! One last little detail and we're good to merge:
The `locales` had meaning when it was used to contain locales but it contain the whole `i18n` configuration. 
Thus I would rename : `src/locales` to `src/i18n`

**Comment by gregoirelacoste** [comment]
@lutangar folder name changed !

**Review by lutangar — APPROVED** [review]
Good job :tada:

**Pull request merged**